### PR TITLE
check_rule_counts metrics ported to Prometheus

### PR DESCRIPTION
**Pull request by FernandoGarzon**

### CODE REVIEW

File: cms/check_rule_counts [hunk 1]
```diff
@@ -21,23 +21,59 @@
...
 if __name__ == '__main__':
+    registry = CollectorRegistry()
+
+    labelnames = ['state']
+    not_ok_labels = ['Not_ok_rules']
+    stuck_cnt_labels = ['Stuck_cnt']
+    replicating_cnt_labels = ['Replicating_cnt']
+
+
+    labelnames.extend(extra_prom_labels.keys())
```
Comment (ericvaandering) [inline cms/check_rule_counts:62]: Make this one count_labels or something specific.

File: cms/check_rule_counts [hunk 2]
```diff
@@ -21,23 +21,59 @@
...
+    not_ok_labels.extend(extra_prom_labels.keys())
+    stuck_cnt_labels.extend(extra_prom_labels.keys())
+    replicating_cnt_labels.extend(extra_prom_labels.keys())
+
+
+    rules_count_gauge = Gauge('rucio_judge_rules_state_count', 'Number of rules in a given state',
```
Comment (ericvaandering) [inline cms/check_rule_counts:68]: This was just rucio_rules_count. Please keep the metric names the same. So what is here with period substituted with underscore and then with whatever mapping/labeling you find in https://github.com/FernandoGarzon/CMSKubernetes/blob/master/kubernetes/rucio/statsd-prometheus-mapping.yaml

File: cms/check_rule_counts [hunk 3]
```diff
@@ -50,61 +86,120 @@ if __name__ == '__main__':
...
         result = (session.query(models.ReplicationRule.state, func.count(models.ReplicationRule.state))
                   .group_by(models.ReplicationRule.state)
                   .with_hint(models.ReplicationRule, 'INDEX_FFS(rules RULES_PK)', 'oracle')
                   .all())
 
+        print(result)
         for state, num in result:
+
+    # Count Rules in x state
             gauge_state = state_map.get(repr(state), 'rules_' + repr(state).lower())
```
Comment (ericvaandering) [inline cms/check_rule_counts:105]: this is much easier. repr(state) can be replaced by str(state.name) and then no substitutions in the next line.

**Review by ericvaandering — CHANGES_REQUESTED** [review]

File: cms/check_rule_counts (hunk 2 above)
Comment (FernandoGarzon) [reply cms/check_rule_counts:68]: I left it as rucio_rules_coutn, but still, I think I'll have to modify the YAML file.

File: cms/check_rule_counts [hunk 4]
```diff
@@ -50,61 +86,120 @@ if __name__ == '__main__':
...
                   .with_hint(models.ReplicationRule, 'INDEX_FFS(rules RULES_PK)', 'oracle')
                   .all())
 
+        print(result)
         for state, num in result:
-            gauge_state = state_map.get(repr(state), 'rules_' + repr(state).lower())
+
+    # Count Rules in x state
+            gauge_state = state_map.get(str(state.name), 'rules_' + str(state.name).lower())
+            #gauge_state = gauge_state.replace("<", "").replace(">", "").split(':')[0]
```
Comment (ericvaandering) [inline cms/check_rule_counts:106]: Please remove commented out code. No one will have any idea why it's there. Then we'll be ready to merge this one.

**Review by ericvaandering — CHANGES_REQUESTED** [review]

**Pull request merged**